app/assets/javascripts/flat.js

```diff
@@ -0,0 +1,39 @@
+document.addEventListener("turbolinks:load", function(){
```

Comment:
This could be refactored further, it's the first pass to get everything abstracted and working